### PR TITLE
support resolve lock and add unit-test

### PR DESCRIPTION
**Pull request by hanfei1991**
Signed-off-by: Han Fei <hanfei19910905@gmail.com>

### CODE REVIEW

**Comment by hanfei1991** [comment]
./rebuild

File: include/pingcap/kv/LockResolver.h [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            resolved.erase(to_remove);
+        }
+    }
+
+    TxnStatus * getResolved(uint64_t txn_id)
+    {
+        auto it = resolved.find(txn_id);
+        if (it == resolved.end())
+            return nullptr;
+        return &(it->second);
```
Comment (disksing) [inline include/pingcap/kv/LockResolver.h]: Is it thread safe to return a reference?

Comment (hanfei1991) [reply include/pingcap/kv/LockResolver.h]: Yes, the pointer is maintained by a global map and will not release until the map is full.

File: include/pingcap/kv/LockResolver.h [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+#include <queue>
+#include <string>
+
+#include <kvproto/tikvpb.grpc.pb.h>
+#include <pingcap/Exception.h>
+#include <pingcap/Log.h>
+#include <pingcap/kv/RegionCache.h>
+
+namespace pingcap
```
[the remark this answers is not in the document]
Comment (hanfei1991) [reply include/pingcap/kv/LockResolver.h]: Any suggestion ? How about use `tikv` as top namespace ?

[the remark this answers is not in the document]
Comment (hanfei1991) [reply include/pingcap/kv/LockResolver.h]: Btw, I suggest to remain it temporarily. We can continue to discuss it when project is released.

**Pull request merged**